builder/builder_test.go

```diff
@@ -36,6 +38,73 @@ func (*queryAll) String() string {
 	return "all"
 }
 
+//nolint:gocritic // tooManyResultsChecker: this function needs to return multiple values for test setup
+func setupTestEnvironment(t *testing.T) (*testapp.App, *genesis.Genesis, *mempool.Pool, *localdb.DB, txstore.TxStore, state.Database, *bfttypes.EventBus) {
+	var chainID monomer.ChainID
+	pool := mempool.New(testutils.NewMemDB(t))
+	blockStore := testutils.NewLocalMemDB(t)
+	txStore := txstore.NewTxStore(testutils.NewCometMemDB(t))
+	ethstatedb := testutils.NewEthStateDB(t)
+
+	app := testapp.NewTest(t, chainID.String())
+	g := &genesis.Genesis{
+		ChainID:  chainID,
+		AppState: testapp.MakeGenesisAppState(t, app),
+	}
+
+	eventBus := bfttypes.NewEventBus()
+	require.NoError(t, eventBus.Start())
+	t.Cleanup(func() {
+		require.NoError(t, eventBus.Stop())
+	})
+
+	require.NoError(t, g.Commit(context.Background(), app, blockStore, ethstatedb))
+
+	return app, g, pool, blockStore, txStore, ethstatedb, eventBus
+}
+
+func createBuilder(pool *mempool.Pool, app *testapp.App, blockStore *localdb.DB, txStore txstore.TxStore, eventBus *bfttypes.EventBus, chainID monomer.ChainID, ethstatedb state.Database) *builder.Builder {
```

Comment:
I feel like it would be cleaner to remove the `createBuilder` func and just inline `builder.New(...)` in the tests